### ISSUE TITLE
Provide a new prop to keep Toast and Layer as normal app nodes to use them with native React 16 portals

### ISSUE THREAD

**Issue by SinHouse** [grommet/grommet#1755]
### Expected Behavior

Toast and Layer should provide a new prop to be rendered as normal app nodes to be able to capture them towards a [React 16 portal.](https://reactjs.org/docs/portals.html)

### Actual Behavior

Toast and Layer are automatically rendered in specific DOM nodes inside the html 'app-root' node and we lose the ability to capture them directly with a portal.

### Html and React code exhibiting the issue

```html
<html lang="en"><head>
    <meta charset="utf-8">
<body>

<!-- Html node used as portal for Toast -->
<div id="message-root"></div>

<div id="app-root">
    <!-- Layer content keep going render here -->
    <div id="skip-link-layer" class="grommet grommetux-layer grommetux-layer--align-top grommetux-layer--hidden">
        <div class="grommetux-layer__container" id="1473245343">
            <a tabindex="-1" aria-hidden="true" style="outline: none;"></a>
        </div>
    </div>
    <div class="grommet grommetux-layer grommetux-layer--align-center">
        <div class="grommetux-layer__container" id="1044976246">
            <a tabindex="-1" aria-hidden="true" style="outline: none;"></a>
        </div>
    </div>
    <!-- Normal Grommet app container -->
    <div lang="es" class="grommet grommetux-app grommetux-app--hidden" aria-hidden="true" style="">
        <span style="display: none;" id="864584761"></span>
        <div class="grommetux-app__announcer" aria-live="polite"></div>
    </div>
</div>

<!-- inside the layer portal just renders an empty span tag-->
<div id="modal-root">
    <span style="display: none;">
</span></div>

<!--
  This HTML file is a template.
-->
<script type="text/javascript" src="/static/js/bundle.js"></script>

</body>
</html>
```
```jsx
type Props = {
    children: React.Node,
}

export default class CustomLayer extends React.Component<Props> {

    render() {
        const layerContent = (
            // If I change Layer by div the complete content is displayed inside the modal-root node
            <Layer>
                {this.props.children}
            </Layer>
        );
        // Portal renders the content in the chosen 'modal-root' div html node
        const modalRoot = document.getElementById('modal-root');
        return ReactDOM.createPortal(layerContent, modalRoot);
    }
}
```
### Proposal
```jsx
// proposal with manual portal management
export default class CustomPortalLayer extends React.Component<Props> {

    render() {
        const layerContent = (
            <Layer usePortal={true}>
                {this.props.children}
            </Layer>
        );
        const modalRoot = document.getElementById('modal-root');
        return ReactDOM.createPortal(layerContent, modalRoot);
    }
}

// proposal with internal Layer code portal management:
export default class CustomPortalLayer extends React.Component<Props> {

    render() {
        return (
            <Layer usePortal='modal-root'>
                {this.props.children}
            </Layer>
        );
    }
}
```
### Your Environment

* React 16.1.1
* Grommet version: 1.9
* Browser Name and version: Chromium 62.0.3202.94
* Operating System and version (desktop or mobile): Ubuntu desktop 16.04.2

**Comment by krizka** [comment]
+1 For this. Now Layer not providing down context, so it is not possible to show Relay components inside modals. Or other context-related things

**Comment by L0ZZI** [comment]
Any chance you want to fashion this into a PR @SinHouse? Maybe @alansouzati has a comment on this?

**Comment by SinHouse** [comment]
Sorry @L0ZZI, I solved the issue changing the pattern in React/Relay data loading and consuming. I think in some context apps it's interesting use native React portals but for now, in our case, the current Layer functionality is enought.

The issue emerged when I tried to load Layer from a table row because I loaded Relay data from it.

I load the relay data from parent table now and I change the controller logic to load the Layer from the table container.
